Lib/Protocols/IdSSLOpenSSL.pas

```diff
@@ -3567,23 +3575,30 @@ procedure TIdSSLContext.InitContext(CtxMode: TIdSSLCtxMode);
   //set SSL Versions we will use
   if HasTLS_method then
   begin
-    if sslvSSLv3 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,SSL3_VERSION)
-    else
-    if sslvTLSv1 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_VERSION)
-    else
-    if sslvTLSv1_1 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_1_VERSION)
-    else
-    if sslvTLSv1_2 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_2_VERSION)
-    else
-    if sslvTLSv1_3 in SSLVersions then
-      SSL_CTX_set_min_proto_version(fContext,TLS1_3_VERSION)
-    else
-    {Set to the lowest now possible}
-      SSL_CTX_set_min_proto_version(fContext,SSL3_VERSION);
+    if SSLVersions <> [] then
+    begin
+      for v := sslvSSLv3 to high(TIdSSLVersions) do
+      begin
+        if v in SSLVersions then
+        begin
+          SSL_CTX_set_min_proto_version(fContext,SSLProtoVersion[v]);
+          break;
+        end;
+     end;
+      for v := high(TIdSSLVersions) downto sslvSSLv3 do

```

Comment:
In Delphi (XE7, 11) compilation fails with "E2008 Incompatible types".
High/Low functions does not work with set types.

Reply:
I have updated the code that defines the "high value" as a constant in order to work around this problem. I don't have Delphi XE7 available, but do have Delphi 2010 still running on an old Windows XP VM. The code compiled and runs OK after the update - so hopefully should also run under XE7. I have updated both branches (old and new lib loaders).